components/brave_extension/extension/brave_extension/content.ts

```diff
@@ -23,3 +23,48 @@ chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
     }
   }
 })
+
+// For dapp detection.
+const script =
+  `
+  function __insertWeb3Installed() {
```

Comment:
@bsclifton It would be good to check this detection logic is also valid for our new crypto wallet extension.